src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -0,0 +1,643 @@
+
+        let theme_manager = event_loop_window_target.theme_manager.clone();
+        let mut window = event_loop_window_target
+            .env
+            .create_window::<ConceptFrame, _>(
+                surface.clone(),
+                Some(theme_manager),
+                (width, height),
+                move |event, mut dispatch_data| {
+                    use sctk::window::{Event, State};
+
+                    let winit_state = dispatch_data.get::<WinitState>().unwrap();
+                    let mut window_updates = winit_state.window_updates.borrow_mut();
+                    let mut window_update = window_updates.get_mut(&window_id).unwrap();
+
+                    match event {
+                        Event::Refresh => {
+                            window_update.refresh_frame = true;
+                        }
+                        Event::Configure { new_size, states } => {
+                            let is_fullscreen = states.contains(&State::Fullscreen);
+                            fullscreen_clone.store(is_fullscreen, Ordering::Relaxed);
+
+                            window_update.refresh_frame = true;
+                            window_update.redraw_requested = true;
+                            if let Some((w, h)) = new_size {
+                                window_update.size = Some(LogicalSize::new(w, h));
+                            }
+                        }
+                        Event::Close => {
+                            window_update.close_window = true;
+                        }
+                    }
+                },
+            )
+            .map_err(|_| os_error!(OsError::WaylandMisc("failed to create window.")))?;
+
+        // Set decorations.
+        if attributes.decorations {
+            window.set_decorate(Decorations::FollowServer);
+        } else {
+            window.set_decorate(Decorations::None);
+        }
+
+        // Min dimensions.
+        let min_size = attributes
+            .min_inner_size
+            .map(|size| size.to_logical::<f64>(scale_factor as f64).into());
+        window.set_min_size(min_size);
+
+        // Max dimensions.
+        let max_size = attributes
+            .min_inner_size
+            .map(|size| size.to_logical::<f64>(scale_factor as f64).into());
+        window.set_max_size(max_size);
+
+        // Set Wayland specific window attributes.
+        if let Some(app_id) = platform_attributes.app_id {
+            window.set_app_id(app_id);
+        }
+
+        // Set common window attributes.
+        //
+        // We set resizable after other attributes, since it touches min and max size under
+        // the hood.
+        window.set_resizable(attributes.resizable);
+        window.set_title(attributes.title);
+
+        // Set fullscreen/maximized if so was requested.
+        match attributes.fullscreen {
+            Some(Fullscreen::Exclusive(_)) => {
+                panic!("Wayland doesn't support exclusive fullscreen.")
```

Comment:
A panic? That seems rather harsh.

Reply:
It was there for a long time so it naturally moved, what would you suggest to do here? Ignore fullscreen request in general or perform a borderless fullscreen? I feel like we can implement exclusive fullscreen on Wayland with viewporter wrt screen sizes, but it won't work with things like setting refresh rate, etc.

Reply:
I think a no-op is better personally.

Reply:
Yeah, winit handles these kind of things rather poorly and inconsistently unfortunately. In general I agree with @JMS55 that a no-op is probably better (and the best) approach.

If it's broken, the user will notice that it doesn't work. That's already enough breakage. We don't need to tear down downstream just because we're mad at them for telling us to do something we cannot do.

Entering a different fullscreen mode is probably a bit confusing and harder to understand when you actually run into the problem, so unless you **must** choose one of the fullscreen modes, I think a no-op is better.

Reply:
>Entering a different fullscreen mode is probably a bit confusing and harder to understand when you actually run into the problem, so unless you must choose one of the fullscreen modes, I think a no-op is better.

I'll add a no-op and a comment that it does nothing on Wayland.